lighthouse-core/gather/driver.js

```diff
@@ -833,6 +833,22 @@ class Driver {
     const promiseArr = urlPatterns.map(url => this.sendCommand('Network.addBlockedURL', {url}));
     return Promise.all(promiseArr);
   }
+
+  /**
+   * Dismiss JavaScript dialogs (alert, confirm, prompt), providing a
+   * generic promptText in case the dialog is a prompt.
+   * @return {!Promise}
+   */
+  dismissJavaScriptDialogs() {
+    return this.sendCommand('Page.enable').then(_ => {
+      this.once('Page.javascriptDialogOpening', _ => {
+        this.sendCommand('Page.handleJavaScriptDialog', {
```

Comment:
if this rejects, this is intentionally unhandled (since we need to resolve as soon as `'Page.enable'` is sent). Should we add a note that that's it's intended for future code readers?